flake.nix

```diff
@@ -0,0 +1,118 @@
+        hyperspace = final.mkYarnPackage {
+          name = packageJson.name;
+          src = ./.;
+          packageJSON = ./package.json;
+          yarnLock = ./yarn2nix.lock;
+        };
+      };
+
+      defaultPackage = forAllSystems (system:
+        (import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }).hyperspace);
+
+      apps = forAllSystems (system: {
+        refresh-dependencies = let pkgs = import nixpkgs { inherit system; };
+        in {
+          type = "app";
+          program = "${
+              pkgs.writeShellScriptBin "refresh-dependencies" ''
+                ${pkgs.nodejs}/bin/npm install --package-lock-only;
+                ${pkgs.nodePackages.yarn}/bin/yarn import;
+                rm package-lock.json;
+                mv yarn.lock yarn2nix.lock;''
+            }/bin/refresh-dependencies";
+        };
+      });
+
+      nixosModules.hyperspace = { config, pkgs, lib, ... }:
+        let cfg = config.services.hyperspace;
+
+        in with lib; {
+          options = {
+            services.hyperspace = {
+              enable = mkEnableOption "hyperspace";
+
+              storage = mkOption {
+                type = types.str;
+                default = "~/.hyperspace";
+                description = "Hyperspace storage directory";
+              };
+
+              host = mkOption {
+                type = types.str;
+                default = "hyperspace";
+                description = "Set unix socket name";
+              };
+
+              port = mkOption {
+                type = types.port;
+                default = 0;
+                description = "Set the port (will use TCP)";
+              };
+
+              no-announce = mkOption {
+                type = types.bool;
+                default = false;
+                description = "Disable all network announces";
+              };
+            };
+          };
+
+          config = mkMerge [
+            ({ nixpkgs.overlays = [ self.overlay ]; })
+
+            (mkIf cfg.enable {
+
+              environment.systemPackages = [ pkgs.hyperspace ];
+
+              systemd.user.services = {
+                hyperspace = {
+                  description = "Hyperspace Daemon";
+                  wantedBy = [ "multi-user.target" ];
+                  after = [ "network.target" ];
+                  serviceConfig = {
+                    ExecStart =
+                      "${pkgs.hyperspace}/bin/hyperspace --storage ${cfg.storage} --host ${cfg.host} --port ${
+                        toString cfg.port
+                      }${if cfg.no-announce then " --no-announce" else ""}";
+                  };
+                };
+              };
+            })
+          ];
+        };
+
+      checks = forAllSystems (system: {
+
+        test = (self.defaultPackage.${system}).overrideAttrs (_: {
+          doBuild = false;
+          doCheck = true;
+          patches = (
+            ./remove-networked-test.patch
```

Comment:
I wish there was a flag to do this. But in leau of that, this seems like the best option. Nice work on setting this up!